aten/src/ATen/TensorIterator.cpp

```diff
@@ -643,6 +643,14 @@ void TensorIteratorBase::coalesce_dimensions() {
   // We can coalesce two adjacent dimensions if either dim has size 1 or if:
   // shape[n] * stride[n] == stride[n + 1].
   auto can_coalesce = [&](int dim0, int dim1) {
+    if (is_reduction_) {
+      for (auto& op : operands_) {
+        if (op.is_output && (op.stride_bytes[dim0] == 0 || op.stride_bytes[dim1] == 0 )) {
+          return false;
```

Comment:
Would this impact the case where we reduce on two dims for which we can actually coalesce?

Reply:
I see some tests are failing due to precision issue. I think it's better to re-formulate p-norm when `input.shape[dim] == 1`.